src/aelin-pool.ts

```diff
@@ -111,13 +111,24 @@ export function handleDealDetail(event: DealDetailEvent): void {
     event.block.timestamp
   );
   dealDetailEntity.isDealFunded = false;
+  dealDetailEntity.totalAmountAccepted = BigInt.fromI32(0);
+  dealDetailEntity.totalWithdrawn = BigInt.fromI32(0);
 
   //get underlyingDealToken symbol and decimals
   const underlyingDealToken = ERC20.bind(event.params.underlyingDealToken);
   dealDetailEntity.underlyingDealTokenSymbol = underlyingDealToken.symbol();
   dealDetailEntity.underlyingDealTokenDecimals = underlyingDealToken.decimals();
   dealDetailEntity.underlyingDealTokenTotalSupply = underlyingDealToken.totalSupply();  
 
+  let dealCreatedEntity = getDealCreated(event.params.dealContract.toHex());
+  if(dealCreatedEntity) {

```

Comment:
nitpick: I generally like specificity here better != null which is the starting state. Otherwise this PR looks great